### PR TITLE
Added tutorials in Harmonic

### PR DESCRIPTION
**Pull request by quarkytale**
Added tutorials from `docs/garden` to `docs/harmonic`.

Unsure about Feature Comparison for Harmonic https://gazebosim.org/docs/garden/comparison, need advice on that.
Also about the target branch, should I be targeting master instead?

### CODE REVIEW

**Review by mabelzhang — COMMENTED** [review]
Re feature comparison - there's an [internal spreadsheet](https://docs.google.com/spreadsheets/d/1mtoziO3mqgiVgPRtmm2pAnZrUkSoL0_qD0G_t-iZUu8/edit#gid=0) about Gazebo roadmap.
There's a public one for Harmonic here https://gazebosim.org/docs/garden/roadmap

I ran a `diff` against the `garden` folder.
I'm pushing a few diffs that I found against the Garden pages that should be updated for Harmonic too.

These files are not yet in this PR so I'll just comment manually:

`harmonic/install.md` needs to be updated with the Ubuntu versions supported - currently says Focal and Jammy, should be Jammy and 24.04 I believe.
(There's also a typo in `garden/install.md`, where it says "uses the Gazebo brand instead of Gazebo", the second Gazebo should be Ignition. That can be in a separate PR later.)

`harmonic/troubleshooting.md` line 54 the dlopen error has both harmonic and libgz-physics6 in it... That looks like it's a find+replace thing from the Garden page. Maybe should just change it back to garden since harmonic is with physics7, and that error message shouldn't be fabricated manually.
Same for line 57.
I'll let you decide on this one.

**Comment by mabelzhang** [comment]
We should definitely at least have the feature comparison.md page copied from Garden. It's not in `index.yaml` right now.
That's just Gazebo-classic to Gazebo comparison. Most of the entries will be the same, with the new things in the spreadsheet added for Harmonic.

**Comment by mabelzhang** [comment]
I pushed a feature comparison page in c557a9c with straight-forward items in the [internal roadmap spreadsheet](https://docs.google.com/spreadsheets/d/1mtoziO3mqgiVgPRtmm2pAnZrUkSoL0_qD0G_t-iZUu8/edit#gid=0).

That is, Wide angle camera, DVL sensor, and Projector for Ogre 1.x.
Re Projector - we should probably create an issue ticket saying projector doesn't work well for Ogre 2 yet. For now, I just link to the original Projector PR.

These other released features I'm not sure whether / where to add to the page:
- We need to add Python somewhere. The only thing existing on there about Python bindings is for SDF. Either in "Others" section or create a new section for it, for better visibility?
- Infra to generate bindings for gz-msgs
- List subscribers in CLI and also get their info
- Compute moments of inertia for SDFormat links (Is this going to make it?)

**Comment by mabelzhang** [comment]
Pushed Ubuntu version in 9143ed7
@azeey to double check - Harmonic is Jammy and 24.04 right?

**Comment by quarkytale** [comment]
> harmonic/troubleshooting.md line 54 the dlopen error has both harmonic and libgz-physics6 in it... That looks like it's a find+replace thing from the Garden page. Maybe should just change it back to garden since harmonic is with physics7, and that error message shouldn't be fabricated manually.
Same for line 57.
Makes sense, reverted the error message back to that of Garden.

**Comment by azeey** [comment]
> @azeey to double check - Harmonic is Jammy and 24.04 right?

Correct.

Regarding where to put some of the new features, if it doesn't make sense to add it here, remember that all the new features will be included in a changelog/release notes that accompanies the release announcement.

File: harmonic/index.yaml [hunk 1]
```diff
@@ -32,3 +32,47 @@ pages:
       - name: troubleshooting
         title: Troubleshooting
         file: troubleshooting.md
+  - name: migration_from_ignition
```
[the remark this answers is not in the document]
Comment (quarkytale) [reply harmonic/index.yaml:35]: I wasn't sure of that too

File: harmonic/index.yaml [hunk 2]
```diff
@@ -32,3 +32,47 @@ pages:
...
+        file: moving_robot.md
+      - name: sdf_worlds
+        title: SDF worlds
+        file: sdf_worlds.md
+      - name: sensors
+        title: Sensors
+        file: sensors.md
+      - name: actors
+        title: Actors
+        file: actors.md
```
[the remark this answers is not in the document]
Comment (quarkytale) [reply harmonic/index.yaml:77]: Can we forward port things in this repo, I think I'll just update the Garden index in this one only

[the remark this answers is not in the document]
Comment (mabelzhang) [reply harmonic/index.yaml:77]: Oh yeah, I forgot this is not in the gz-sim repo 😅 You're right. Then it's even better!

File: harmonic/index.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mabelzhang) [reply harmonic/index.yaml:35]: @azeey says we use the pre-release repos for the tutorial party. So we should add all those 3 binary install pages.

[the remark this answers is not in the document]
Comment (mabelzhang) [reply harmonic/index.yaml:35]: E.g. https://github.com/gazebosim/docs/pull/310/files
Probably looking at the history of the Mac and Windows binary install files in Garden will tell you how to do the pre-release for those.
I'm out and about today and in meetings so I can't do this right now...

File: harmonic/index.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (quarkytale) [reply harmonic/index.yaml:77]: Fixed in [08dad38](https://github.com/gazebosim/docs/pull/372/commits/08dad387133e0d154a28951057f3497cc55294c9)

File: harmonic/index.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (quarkytale) [reply harmonic/index.yaml:35]: No worries, I can update that

File: harmonic/index.yaml [hunk 3]
```diff
@@ -32,3 +41,47 @@ pages:
       - name: troubleshooting
         title: Troubleshooting
         file: troubleshooting.md
+  - name: migration_from_ignition
+    title: Migration from Ignition
+    file: migration_from_ignition.md
+    description: Migration from the Ignition name
+
+  - name: tutorials
```
Comment (azeey) [inline harmonic/index.yaml:49]: We're missing the `comparison` key here right?
```
  - name: comparison
    title: Feature Comparison
    file: comparison.md
    description: A list of features present in Gazebo-classic and the status of their migration to Gazebo.
```

File: harmonic/install.md
Comment (azeey) [inline harmonic/install.md:1]: The "Binary installation" section says there are no binaries at the moment.

File: harmonic/install_osx.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Binary Installation on MacOS
+
+All the Harmonic binaries are available in Catalina and BigSur using the
```
Comment (azeey) [inline harmonic/install_osx.md:3]:
```suggestion
All the Harmonic binaries are available in Big Sur and Monterey using the
```

File: harmonic/install_osx_src.md
Comment (azeey) [inline harmonic/install_osx_src.md:1]: Remove Catalina

**Review by azeey — COMMENTED** [review]
There are some changes that are in Garden, but not in the Harmonic directory. 
examples:
* Warning at the top of the windows install page
* A sentence about `MESA_GL_VERSION_OVERRIDE` in troubleshooting.md

Can you do a diff and see if there are other changes?

**Comment by mabelzhang** [comment]
Can we do a quick fix on this?
> There's also a typo in garden/install.md, where it says "uses the Gazebo brand instead of Gazebo", the second Gazebo should be Ignition. That can be in a separate PR later.

**Comment by mabelzhang** [comment]
@azeey Looking from the roadmap spreadsheet, is it correct to say there are Python bindings for gz-physics and gz-sim in Harmonic? (gz-math and sdformat are from previous releases)
Any other libraries?
I'm adding a section in the comparison page to just talk about python bindings.

**Comment by azeey** [comment]
There are python bindings for gz-transport and gz-sim. We should also have the ability to write Gazebo systems using python. That being said, I think it would be best to merge this PR and add that section in a follow up because we need the Harmonic directory created for [yatm](https://github.com/audrow/yatm) to find them and generate corresponding issues for the tutorial party.

**Comment by mabelzhang** [comment]
Okay. In that case, I don't have anything else to commit.
I'll do another review after I'm back from another meeting.

**Review by azeey — APPROVED** [review]

**Pull request merged**